src/BenchmarkDotNet/Running/WakeLock.cs

```diff
@@ -0,0 +1,44 @@
+﻿using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Detectors;
+using BenchmarkDotNet.Helpers;
+using System;
+using System.Linq;
+
+namespace BenchmarkDotNet.Running;
+
+internal partial class WakeLock
+{
+    public static WakeLockType GetWakeLockType(BenchmarkRunInfo[] benchmarkRunInfos) =>
+        benchmarkRunInfos.Select(static i => i.Config.WakeLock).Max();
+
+    public static IDisposable Request(WakeLockType wakeLockType, string reason) =>
+        wakeLockType == WakeLockType.No || !OsDetector.IsWindows() ? null : new WakeLockSentinel(wakeLockType, reason);
```

Comment:
What is the minimum windows version for this API?

Reply:
Power(Create|Set|Clear)Request are [>= Windows 7;Windows Server 2008 R2 ](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-powercreaterequest#requirements)

CloseHandle is >= Windows 2000 😉

Reply:
Can you add the windows version check like in TaskbarProgress?